packages/typespec-python/generator/pygen/codegen/templates/model_base.py.jinja2

```diff
@@ -675,6 +677,8 @@ def _get_deserialize_callable_from_annotation(  # pylint: disable=R0911, R0915,
     rf: typing.Optional["_RestField"] = None,
 ) -> typing.Optional[typing.Callable[[typing.Any], typing.Any]]:
     if not annotation or annotation in [int, float]:
+        if annotation is int and rf and rf._format == "str":
```

Comment:
i don't think we have to modify `model_base.py`, we can pass in a `deserialization` function into the `rest_field` and have it serialize and deserialize as a string

Reply:
https://github.com/Azure/autorest.python/blob/848048796843fb226390077e42549350de22f8a4/packages/typespec-python/generator/pygen/codegen/templates/model_base.py.jinja2#L829-L840
Appreciate if you could explain more how to pass in `deserialization` function? It seems that RestField has no parameter to pass it.

Reply:
you pass it through the `type` kwarg. For example, in [this](https://github.com/Azure/autorest.python/blob/main/packages/typespec-python/test/generic_mock_api_tests/unittests/test_model_base_serialization.py#L953) test, we've typed the prop as being a sequence, but in terms of serislizstion / deserialization, we want it to be a `set`, so we pass in a callback. We should be able to do the same thing here, where we type it as an int, but we pass in `type=lambda x: str(x)` / write a callback that does that instead

Reply:
for datetime/bytes encoding, we put the logic in `model_base.py`. shall we keep consistent for all the encoding things? another thing is if use callback, how to deal with serialization?

Reply:
@iscai-msft  I think it is better not handle this feature with call_back so that we could keep consistent for all encoding things. But your comment hint me that we could confirm the deserialize and serialize function for model property when generate code so that we could save time for model initialization and I have created https://github.com/Azure/autorest.python/issues/2797 to track it.

Reply:
Since the model is vendored we can def change things, so I think we can have the rest field take in `serialization` and `deserialization` as callbacks. As @msyyc said we've already thought of passing explicit callbacks into rest field for perf improvements, so I think what we should do is do a slight redesign where we have `serialization` and `deserialization` callback inputs. When we're generating, we can generate with explicit callback inputs so we don't have to use perf time to determine which serialize / deserialize function we should use

Reply:
Exactly. And the change of redesign for `serialization`/`deserialization` will be a little huge so I would like to do it in another PR.

Reply:
Ok I've added a separate issue for the `serialization` and `deserialization`, https://github.com/Azure/autorest.python/issues/2801